azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/StorageService.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.domain;
+
+import com.google.auto.value.AutoValue;
+import java.util.Date;
+import java.util.Map;
+
+import com.google.common.collect.ImmutableMap;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+
+import org.jclouds.azurecompute.arm.util.GetEnumValue;
+
+@AutoValue
+public abstract class StorageService {
+
+   public enum AccountType {
+
+      Standard_LRS,
+      Standard_ZRS,
+      Standard_GRS,
+      Standard_RAGRS,
+      Premium_LRS,
+      UNRECOGNIZED;
+
+       public static AccountType fromValue(final String text) {
+           return (AccountType) GetEnumValue.fromValueOrDefault(text, AccountType.UNRECOGNIZED);
+       }
+   }
+
+   public enum RegionStatus {
+
+       Available,
+       Unavailable,
+       UNRECOGNIZED;
+
+       public static RegionStatus fromValue(final String text) {
+           return (RegionStatus) GetEnumValue.fromValueOrDefault(text, RegionStatus.UNRECOGNIZED);
+       }
+
+   }
+
+   public enum Status {
+
+       Creating,
+       Created,
+       Deleting,
+       Deleted,
+       Changing,
+       ResolvingDns,
+       Succeeded,
+       UNRECOGNIZED;
+
+       public static Status fromValue(final String text) {
+           return (Status) GetEnumValue.fromValueOrDefault(text, Status.UNRECOGNIZED);
+       }
+   }
+
+   @AutoValue
+   public abstract static class StorageServiceProperties {
+
+       /**
+        * Specifies whether the account supports locally-redundant storage, geo-redundant storage, zone-redundant
+        * storage, or read access geo-redundant storage.
+        */
+       public abstract AccountType accountType();
+
+       /**
+        * Specifies the time that the storage account was created.
+        */
+       public abstract Date creationTime();
+
+       /**
+        * Specifies the endpoints of the storage account.
+        */
+       @Nullable
```

Comment:
@nacx 
So I did remove this annotation in this class. But in order to support create and update requests without tags, I added `@Nullable` in the [StorageAccountApi](https://github.com/ritazh/jclouds-labs/blob/azurecomputearm-storageaccountapi/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/features/StorageAccountApi.java#L90).